src/tx_xform_prv.erl

```diff
@@ -101,14 +103,25 @@ pp(Forms, F) ->
 
 clauses(Found, L0) ->
     {Cs, _} = lists:mapfoldl(
-                fun({Mod, _, {RecName, _}}, L) ->
+                fun({Mod, _, {RecName, _}, _}, L) ->
                         {{clause, L, [{record, L, RecName, []}],
                           [],
                           [{atom, L+1, Mod}]},
                          L+2}
                 end, L0, Found),
     Cs.
 
+type_clauses(Found, L0) ->
+    {Cs, _} = lists:mapfoldl(
+                fun({Mod, _, _, Str}, L) ->
```

Comment:
I think you need to match on the case where `Str` is `{error, Reason}` (see line 214).
You could e.g. catch exceptions at the top of the codegen and let the plugin fail, or simply insert an error tuple instead of the `clause` tuple. This would lead to a fairly readable compilation error, rather than a codegen crash with a nasty stack trace.